frontend/public/components/utils/file-input.tsx

```diff
@@ -182,7 +183,7 @@ export const DroppableFileInput = withDragDropContext(
           errorMessage={this.state.errorMessage}
           onDrop={this.handleFileDrop}
           onChange={this.onDataChange}
-          inputFileData={this.state.inputFileData}
+          inputFileData={this.state.inputFileData as string}
```

Comment:
since `DroppableFileInput` uses `reader.readAsText` we can be more specific with the type of state - `inputFileData: string;` instead of `inputFileData: string | ArrayBuffer;` and then you wont have to cast to string here

Reply:
nice !